Manage Organizations doesn't work

When I use the Manage Organizations button, the new organization doesn't get written to the database. Without an organization in the database, you can't go to the next page when creating a project.

When trying to create a new organization, I get an error on the logo. The log shows:
* [ERROR] fastapi | create_organization:74 | [Errno 13] Permission denied: 'app/images/sslogo2.jpg'. This prevents creating an organization from working. I have to manually add an org to the database to be able to create a project.

Note that the recent updates to the project endpoints/logic has broken the tests we have:
https://github.com/hotosm/fmtm/actions/runs/5344620635/jobs/9689213226

```
---------------------------- Captured log teardown -----------------------------
DEBUG    app.main:main.py:134 Shutting down FastAPI server.
=========================== short test summary info ============================
FAILED tests/test_projects.py::test_list_projects_summuries - assert 400 == 200
 +  where 400 = <Response [400]>.status_code
 +  and   200 = status.HTTP_200_OK
FAILED tests/test_projects.py::test_list_organisations - assert 1 == 3
 +  where 1 = len({'errors': [{'error': "value is not a valid integer['path', 'project_id']", 'loc': ['path', 'project_id'], 'msg': 'value is not a valid integer'}]})
 +    where {'errors': [{'error': "value is not a valid integer['path', 'project_id']", 'loc': ['path', 'project_id'], 'msg': 'value is not a valid integer'}]} = <bound method Response.json of <Response [400]>>()
 +      where <bound method Response.json of <Response [400]>> = <Response [400]>.json
FAILED tests/test_projects.py::test_list_projects_features - assert 400 == 200
 +  where 400 = <Response [400]>.status_code
 +  and   200 = status.HTTP_200_OK
FAILED tests/test_projects.py::test_project_logs - assert 400 == 200
 +  where 400 = <Response [400]>.status_code
 +  and   200 = status.HTTP_200_OK
FAILED tests/test_projects.py::test_categories - assert 404 == 200
 +  where 404 = <Response [404]>.status_code
 +  and   200 = status.HTTP_200_OK
FAILED tests/test_projects.py::test_get_task - assert 422 == 200
 +  where 422 = <Response [422]>.status_code
 +  and   200 = status.HTTP_200_OK
FAILED tests/test_projects.py::test_create_project - TypeError: unhashable type: 'dict'
FAILED tests/test_projects.py::test_delete_project - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_upload_project - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_upload_xlsform - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_upload_multi_polygon - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_upload_boundary - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_download_project - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_download_task - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_generate_files - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_create_organization - TypeError: unhashable type: 'dict'
FAILED tests/test_projects.py::test_preview_task - TypeError: Object of type set is not JSON serializable
FAILED tests/test_projects.py::test_add_features - TypeError: Object of type set is not JSON serializable
18 failed, 3 passed in 3.28s
Error: Process completed with exit code 1.
```

--- a/src/backend/app/organization/organization_crud.py
+++ b/src/backend/app/organization/organization_crud.py
@@ -60,7 +60,31 @@ async def get_organisation_by_name(db: Session, name: str):
     return db_organisation
 
 
-async def create_organization(db: Session, name: str, description: str, url: str, logo: UploadFile = File(...)):
+async def upload_image(db: Session, file: UploadFile(None)):
+    # Check if file with the same name exists
+    filename = file.filename
+    file_path = f"{IMAGEDIR}{filename}"
+    while os.path.exists(file_path):
+
+        # Generate a random character
+        random_char = ''.join(random.choices(string.ascii_letters + string.digits, k=3))
+
+        # Add the random character to the filename
+        logo_name, extension = os.path.splitext(filename)
+        filename = f"{logo_name}_{random_char}{extension}"
+        file_path = f"{IMAGEDIR}{filename}"
+
+    # Read the file contents
+    contents = await file.read()
+
+    # Save the file
+    with open(file_path, "wb") as f:
+        f.write(contents)
+
+    return filename
+
+
+async def create_organization(db: Session, name: str, description: str, url: str, logo: UploadFile(None)):
     """
     Creates a new organization with the given name, description, url, type, and logo.
     Saves the logo file to the app/images folder.
@@ -79,30 +103,23 @@ async def create_organization(db: Session, name: str, description: str, url: str
 
     # create new organization
     try:
+        logo_name = await upload_image(db, logo) if logo else None
+
         db_organization = db_models.DbOrganisation(
             name=name,
             slug=generate_slug(name),
             description=description,
-            url=url
+            url=url,
+            logo=logo_name
         )
 
-        # Save logo file to app/images folder
-        # file_path = os.path.join("app/images")
-        # if not os.path.exists(file_path):
-        #     os.makedirs(file_path)
-
-        # with open(os.path.join(file_path, logo.filename), "wb") as file:
-        #     file.write(await logo.read())
-
-        db_organization.logo = logo.filename if logo else None
         db.add(db_organization)
         db.commit()
         db.refresh(db_organization)
     except Exception as e:
         logger.error(e)
         raise HTTPException(
-            status_code=400, detail=f" ----- Error: {e}"
+            status_code=400, detail=f"Error creating organization: {e}"
         ) from e
 
     return True
-

--- a/src/backend/app/organization/organization_crud.py
+++ b/src/backend/app/organization/organization_crud.py
@@ -16,6 +16,8 @@
 #     along with FMTM.  If not, see <https:#www.gnu.org/licenses/>.
 #
 import os
+import random
+import string
 from fastapi import HTTPException, File,UploadFile
 from fastapi.logger import logger as logger
 import re
@@ -24,6 +26,7 @@ from sqlalchemy.orm import Session
 
 from ..db import db_models
 
+IMAGEDIR = "app/images/"
 
 # --------------
 # ---- CRUD ----

